syntax = "proto3";

package TW.Aeternity.Proto;
option java_package = "wallet.core.jni.proto";

// Input data necessary to create a signed transaction.
message SigningInput {
    string from_address = 1;
    string to_address = 2;
    uint64 amount = 3;
    uint64 fee = 4;
    string payload = 5; //optional
    uint64 ttl = 6;
    uint64 nonce = 7;
    bytes private_key = 8;
}

// Transaction signing output.
message SigningOutput {
    // Signed and encoded transaction bytes.
    string encoded = 1;
    // Signature.
    string signature = 2;
}
